lib/customer.rb

```diff
@@ -0,0 +1,36 @@
+# require_relative 'data/customers.csv'
+require 'csv'
+
+class Customer
+  attr_reader :id
+  attr_accessor :email, :address, :all_customers
+
+  def initialize(id = 0, email = '', address = {})
+    @id = id
+    @email = email
+    @address = address
+  end
+
+  def self.all
+    all_customers = []
+    CSV.read('../data/customers.csv').each do |row|
```

Comment:
The path you have here makes it so that `rake` won't successfully run tests.  (They work when run individually.)

To make `rake` work you can update the path relative to the project root:
```suggestion
    CSV.read('data/customers.csv').each do |row|
```

It's really unfortunate that this works differently from `require_relative`.